### PR TITLE
Support Forge

### PR DESCRIPTION
**Pull request by babedev**
**What's in this PR?**

Introduced Forge deserialisation for Forge users.

**Usage**

```Kotlin
data class HttpBinUserAgentModel(var userAgent: String = "", var status: String = "")

private val httpBinUserDeserializer = { json: JSON ->
    ::HttpBinUserAgentModel.create.
                map(json at "userAgent").
                apply(json at "status")
}

Fuel.get("/user-agent").responseObject(httpBinUserDeserializer) { _, _, result -> }
```

### CODE REVIEW

File: fuel-forge/src/test/kotlin/com/github/babedev/fuel/forge/FuelForgeTest.kt [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    @Test
+    fun testProcessingGenericList() {
+        Fuel.get("https://api.github.com/repos/kittinunf/Fuel/issues").responseObjects(issueInfoDeserializer) { _, _, result ->
+            val issues = result.get()
+            Assert.assertNotEquals(issues.size, 0)
+            Assert.assertThat(issues[0], CoreMatchers.isA(IssueInfo::class.java))
+        }
+    }
+}
```
Comment (kittinunf) [inline fuel-forge/src/test/kotlin/com/github/babedev/fuel/forge/FuelForgeTest.kt]: Can you remove the new line at the end of the file?

Comment (babedev) [reply fuel-forge/src/test/kotlin/com/github/babedev/fuel/forge/FuelForgeTest.kt]: Fixed 😎

**Comment by kittinunf** [comment]
I think we can merge this, @babedev can you please help rebase against latest master.

**Comment by babedev** [comment]
Roger, I will rebase and merge it.

**Comment by babedev** [comment]
I have done something wrong, so I will close this and create another PR.

**Pull request closed without merging**